composition/src/listener_component.cpp

```diff
@@ -23,16 +23,31 @@
 namespace composition
 {
 
+/**
+ * Create a Listener "component" that subclasses the generic rclcpp::Node base class.
+ * Components get built into shared libraries and as such do not write their own main functions.
+ * The process using the component's shared library will instantiate the class as a ROS node.
+ */
 Listener::Listener()
 : Node("listener")
 {
+  /**
+    * Create a callback function for when messages are received.
+    * Variations of this function also exist using, for example, UniquePtr for zero-copy transport.
+    */
```

Comment:
Should these be line comments rather then block comments. Doxygen (`/**`) would also not extract those I assume?